buffer/src/test/java/io/netty/buffer/PoolArenaTest.java

```diff
@@ -43,6 +44,27 @@ public void testNormalizeAlignedCapacity() throws Exception {
         }
     }
 
+    @Test
+    public void testDirectArenaOffsetCacheLine() throws Exception {
+        int capacity = 5;
+        int alignment = 128;
+
+        for (int i = 0; i < 100000; i++) {
+            ByteBuffer bb = PlatformDependent.useDirectBufferNoCleaner()
+                    ? PlatformDependent.allocateDirectNoCleaner(capacity + alignment)
+                    : ByteBuffer.allocateDirect(capacity + alignment);
+
+            PoolArena.DirectArena arena = new PoolArena.DirectArena(null, 0, 0, 9, 9, alignment);
+            int offset = arena.offsetCacheLine(bb);
+
+            long address = PlatformDependent.directBufferAddress(bb);
+            int expected = alignment - (int) (address & (alignment - 1));
```

Comment:
Can we somehow change the test that we not duplicate just the logic here but ensure the correct one is set in the chunk. Or at least hardcore the number here that we expect?

Reply:
Since address is determined by the runtime dynamically, it seems unlikely to hardcode the offset number we expect here. We can't make sure the correct one is set if we  don't re-calculate based on the address.

Any ideas? :)

Reply:
True... Maybe we could just use a modulo operation here tho to proof that what we do with our bitwise tricks is correct ?

Reply:
What about this Assert.assertEquals(0, (offset + address) & (alignment - 1))?

Reply:
yeah I think thats better es well.